lib/ansible/module_utils/basic.py

```diff
@@ -2603,7 +2603,13 @@ def run_command(self, args, check_rc=False, close_fds=True, executable=None, dat
 
             stdout = b''
             stderr = b''
-            selector = selectors.DefaultSelector()
+            try:
+                selector = selectors.DefaultSelector()
+            except OSError:
+                # Failed to detect default selector for the given platform
+                # Set it based upon 'select'
+                selector = selectors.SelectSelector()
+
```

Comment:
Maybe we could do something like this to let us determine the our own precedence order.
```suggestion
            for s in ['KqueueSelector', 'DevpollSelector', 'EpollSelector', 'PollSelector', 'SelectSelector']:
                try:
                    selector = getattr(selectors, s)()
                    break
                except (OSError, AttributeError):
                    continue
```

This is the same precedence order in `selectors`, but we could change it. Maybe this fix should be in `selectors` rather than here. Open to feedback.